packages/fxa-auth-server/lib/payments/stripe.ts

```diff
@@ -1033,6 +1052,16 @@ export class StripeHelper {
     if (!this.paidInvoice(subscription.latest_invoice)) {
       throw error.paymentFailed();
     }
+
+    country = this.extractSourceCountryFromSubscription(subscription);
```

Comment:
We create subscriptions with payment intents, and sometimes setup intents, these are not always located on the subscription at this state of subscription setup. Sometimes the payment intent fails, then a setup intent is run later against the newly created subscription to finalize payment.